From 28ac1a2cbafeb5bd4050f253a8ce93601c949a7c Mon Sep 17 00:00:00 2001
From: Dorota Czaplejewicz <dorota.czaplejewicz@puri.sm>
Date: Wed, 17 Oct 2018 12:51:07 +0000
Subject: [PATCH 03/11] imwayland.c: fix formatting

---
 modules/input/imwayland.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules/input/imwayland.c b/modules/input/imwayland.c
index 2e06f6fb88..7631ae955d 100644
--- a/modules/input/imwayland.c
+++ b/modules/input/imwayland.c
@@ -619,7 +619,7 @@ registry_handle_global (void               *data,
                           &zwp_text_input_manager_v3_interface, 1);
       global->text_input =
         zwp_text_input_manager_v3_get_text_input (global->text_input_manager,
-                                          gdk_wayland_seat_get_wl_seat (seat));
+                                                  gdk_wayland_seat_get_wl_seat (seat));
       global->serial = 0;
       zwp_text_input_v3_add_listener (global->text_input,
                                       &text_input_listener, global);
@@ -636,8 +636,8 @@ registry_handle_global_remove (void               *data,
   if (id != global->text_input_manager_wl_id)
     return;
 
-  g_clear_pointer(&global->text_input, zwp_text_input_v3_destroy);
-  g_clear_pointer(&global->text_input_manager, zwp_text_input_manager_v3_destroy);
+  g_clear_pointer (&global->text_input, zwp_text_input_v3_destroy);
+  g_clear_pointer (&global->text_input_manager, zwp_text_input_manager_v3_destroy);
 }
 
 static const struct wl_registry_listener registry_listener = {
-- 
2.20.0

