diff --git a/include/Rivet/Tools/Utils.hh b/include/Rivet/Tools/Utils.hh
--- a/include/Rivet/Tools/Utils.hh
+++ b/include/Rivet/Tools/Utils.hh
@@ -101,7 +101,7 @@ namespace Rivet {
   /// Convert a string to lower-case
   inline string toLower(const string& s) {
     string out = s;
-    std::transform(out.begin(), out.end(), out.begin(), (int(*)(int)) tolower);
+    std::transform(out.begin(), out.end(), out.begin(), (int(*)(int)) ::tolower);
     return out;
   }
 
@@ -109,7 +109,7 @@ namespace Rivet {
   /// Convert a string to upper-case
   inline string toUpper(const string& s) {
     string out = s;
-    std::transform(out.begin(), out.end(), out.begin(), (int(*)(int)) toupper);
+    std::transform(out.begin(), out.end(), out.begin(), (int(*)(int)) ::toupper);
     return out;
   }
 
