From aefbcf8029686bde94bfdcc2b5599f9a8a2a421b Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Tue, 17 Oct 2017 22:57:02 -0500
Subject: [PATCH] use fenv instead of non-standard fpu_control

---
 core/Main.cc | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/core/Main.cc b/core/Main.cc
index 2b0d97b..8ad95fb 100644
--- a/core/Main.cc
+++ b/core/Main.cc
@@ -78,8 +78,12 @@ int main(int argc, char** argv)
         // printf("This is MiniSat 2.0 beta\n");
         
 #if defined(__linux__)
-        fpu_control_t oldcw, newcw;
-        _FPU_GETCW(oldcw); newcw = (oldcw & ~_FPU_EXTENDED) | _FPU_DOUBLE; _FPU_SETCW(newcw);
+        fenv_t fenv;
+
+        fegetenv(&fenv);
+        fenv.__control_word &= ~0x300; /* _FPU_EXTENDED */
+        fenv.__control_word |= 0x200; /* _FPU_DOUBLE */
+        fesetenv(&fenv);
         printf("WARNING: for repeatability, setting FPU to use double precision\n");
 #endif
         // Extra options:
-- 
2.14.2

