From b352c711f6a9991f6e154a54af8b45a045284e9c Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Sat, 6 Apr 2019 14:01:13 -0500
Subject: [PATCH] quoting

---
 contrib/vdir/calcurse-vdirsyncer | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/contrib/vdir/calcurse-vdirsyncer b/contrib/vdir/calcurse-vdirsyncer
index c5371b5..bef8e76 100755
--- a/contrib/vdir/calcurse-vdirsyncer
+++ b/contrib/vdir/calcurse-vdirsyncer
@@ -66,6 +66,6 @@ while [ $# -gt 0 ]; do
 	esac
 done
 
-calcurse-vdir export "$VDIR" -D "$DATADIR" "$FORCE" "$VERBOSE" && \
+calcurse-vdir export "$VDIR" -D "$DATADIR" $FORCE $VERBOSE && \
 	vdirsyncer sync && \
-	calcurse-vdir import "$VDIR" -D "$DATADIR" "$FORCE" "$VERBOSE"
+	calcurse-vdir import "$VDIR" -D "$DATADIR" $FORCE $VERBOSE
-- 
2.21.GIT

