--- config.def.h	2013-04-20 13:29:39.000000000 +0000
+++ config.h	2013-05-10 13:49:24.529417690 +0000
@@ -5,7 +5,7 @@
  *
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
-static char font[] = "Liberation Mono:pixelsize=12:antialias=false:autohint=false";
+static char font[] = "Inconsolata:pixelsize=14:style=medium:antialias=true,PowerlineSymbols";
 static int borderpx = 2;
 static char shell[] = "/bin/sh";
 
@@ -20,6 +20,12 @@
 static unsigned int xfps = 60;
 static unsigned int actionfps = 30;
 
+/*
+* blinking timeout (set to 0 to disable blinking) for the terminal blinking
+* attribute.
+*/
+static unsigned int blinktimeout = 800;
+
 /* TERM value */
 static char termname[] = "st-256color";
 
@@ -28,50 +34,51 @@
 
 /* Terminal colors (16 first used in escape sequence) */
 static const char *colorname[] = {
-	/* 8 normal colors */
-	"black",
-	"red3",
-	"green3",
-	"yellow3",
-	"blue2",
-	"magenta3",
-	"cyan3",
-	"gray90",
-
-	/* 8 bright colors */
-	"gray50",
-	"red",
-	"green",
-	"yellow",
-	"#5c5cff",
-	"magenta",
-	"cyan",
-	"white",
-
-	[255] = 0,
-
-	/* more colors can be added after 255 to use with DefaultXX */
-	"#cccccc",
+  /* 8 normal colors */
+  "#073642",
+  "#dc322f",
+  "#859900",
+  "#b58900",
+  "#268bd2",
+  "#d33682",
+  "#2aa198",
+  "#eee8d5",
+
+  /* 8 bright colors */
+  "#002b36",
+  "#cb4b16",
+  "#586e75",
+  "#657b83",
+  "#839496",
+  "#6c71c4",
+  "#93a1a1",
+  "#fdf6e3",
+
+  [255] = 0,
+
+  /* more colors can be added after 255 to use with DefaultXX */
+  "#cccccc",
+  "#333333",
 };
 
 
 /*
  * Default colors (colorname index)
- * foreground, background, cursor
+ * foreground, background, cursor, unfocused cursor
  */
-static unsigned int defaultfg = 7;
-static unsigned int defaultbg = 0;
-static unsigned int defaultcs = 256;
+static unsigned int defaultfg = 12;
+static unsigned int defaultbg = 8;
+static unsigned int defaultcs = 14;
 
 /*
  * Colors used, when the specific fg == defaultfg. So in reverse mode this
  * will reverse too. Another logic would only make the simple feature too
  * complex.
  */
-static unsigned int defaultitalic = 11;
-static unsigned int defaultunderline = 7;
+static unsigned int defaultitalic = 10;
+static unsigned int defaultunderline = 14;
 
-/* Internal shortcuts. */
+/* Internal keyboard shortcuts. */
 #define MODKEY Mod1Mask
 
 static Shortcut shortcuts[] = {
