--- swift/utils/build-script-impl	2017-01-24 11:02:03.697555830 -0600
+++ swift-pax/utils/build-script-impl	2017-01-24 11:02:26.773939163 -0600
@@ -1755,6 +1755,16 @@ function set_swiftpm_bootstrap_command()
     fi
 }
 
+## XXX: Taken from nixpkgs /pkgs/stdenv/generic/setup.sh
+isELF() {
+    local fn="$1"
+    local magic
+    exec {fd}< "$fn"
+    read -n 4 -u $fd magic
+    exec {fd}<&-
+    if [[ "$magic" =~ ELF ]]; then return 0; else return 1; fi
+}
+
 #
 # Configure and build each product
 #
@@ -2541,6 +2551,12 @@ for host in "${ALL_HOSTS[@]}"; do
             fi
 
             call ${DISTCC_PUMP} "${CMAKE}" --build "${build_dir}" $(cmake_config_opt ${product}) -- "${BUILD_ARGS[@]}" ${build_targets[@]}
+            
+						while IFS= read -r -d $'\0' i; do
+								if ! isELF "$i"; then continue; fi
+								echo "setting pax flags on $i"
+								paxctl -czexm "$i" || true
+						done < <(find "${build_dir}" -executable -type f -wholename "*/bin/*" -print0)
         fi
     done
 done
