diff --git a/Configure b/Configure
index af2d6c0..5064cd0 100755
--- a/Configure
+++ b/Configure
@@ -3251,6 +3251,7 @@ if test -f config.sh; then
 		;;
 	esac
 fi
+myuname=`uname -s | tr A-Z a-z`" localhost 3.0.0 #1-nixos smp thu jan 1 00:00:01 utc 1970 "`echo "$stdenv_system" | tr - \  `
 . ./UU/checkcc
 if test ! -f config.sh; then
 	$cat <<EOM
@@ -3763,8 +3764,11 @@ none)  osvers='' ;;
 
 . ./posthint.sh
 
 : who configured the system
 cf_time=`LC_ALL=C; LANGUAGE=C; export LC_ALL; export LANGUAGE; $date 2>&1`
+cf_time='Thu Jan  1 00:00:01 UTC 1970'
+
+cf_by=noone
 case "$cf_by" in
 "")
 	cf_by=`(logname) 2>/dev/null`
@@ -9049,6 +9056,8 @@ case "$mydomain" in
 	mydomain=`echo $mydomain | ./tr '[A-Z]' '[a-z]'`
 	;;
 esac
+myhostname="nomachine"
+mydomain=".nixos.org"
 
 : a little sanity check here
 case "$phostname" in
@@ -9115,6 +9124,7 @@ while test "$cont"; do
 		;;
 	esac
 done
+cf_email="noone@nixos.org"
 
 : Ask e-mail of administrator
 $cat <<EOM
