From cc94388f678ea3b3c6dd4ec27f0d7e3e782e2bc1 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Wed, 17 Apr 2019 22:49:41 -0500
Subject: [PATCH] fix completions

---
 git-extra-commands.plugin.zsh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/git-extra-commands.plugin.zsh b/git-extra-commands.plugin.zsh
index bb9dacd..d7a054c 100644
--- a/git-extra-commands.plugin.zsh
+++ b/git-extra-commands.plugin.zsh
@@ -22,7 +22,7 @@ alias gitroot='cd $(git rev-parse --show-toplevel) && echo "$_"'
 #   incoming-commits  (appears to be a dupe of grab)
 #   mark-all-resolved (git: 'conflicts' is not a git command.)
 zstyle ':completion:*:*:git:*' user-commands \
-  age:'A git-blame viewer, written using PyGTK written by Kristoffer Gronlund ' \
+  age:'A git-blame viewer, written using PyGTK written by Kristoffer Gronlund' \
   big-file:'List disk size of files in ref' \
   change-author:'Rewrite commits, updating author/email' \
   changes:'List authors/emails with commit count' \
@@ -40,7 +40,7 @@ zstyle ':completion:*:*:git:*' user-commands \
   improved-merge:'Sophisticated git merge with integrated CI check and automatic cleanup upon completion' \
   incoming:'Fetch remote tracking branch, and list incoming commits' \
   ls-object-refs:'Find references to <object> SHA1 in refs, commits, and trees. All of them' \
-  maildiff: 'Simple git command to email diff in color to reviewer/co-worker & optionally attach patch file.' \
+  maildiff:'Simple git command to email diff in color to reviewer/co-worker & optionally attach patch file.' \
   maxpack:'Repack with maximum compression' \
   nuke:'Nukes a branch locally and on the origin remote' \
   object-deflate:'Deflate an loose object file and write to standard output' \
-- 
2.21.GIT

