diff --git a/configure.ac b/configure.ac
index 89ef9398e..89b5c4e0b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -853,8 +853,7 @@ if test x"$LIBUSB_NAME" != x ; then
                   have_libusb=yes ])
    AC_MSG_CHECKING([libusb include dir])
    usb_incdir_found="no"
-   for _incdir in "" "/usr/include/libusb-1.0" \
-       "/usr/local/include/libusb-1.0" "/usr/pkg/include/libusb-1.0"; do
+   for _incdir in "$($PKG_CONFIG --variable=includedir libusb-1.0)/libusb-1.0"; do
      _libusb_save_cppflags=$CPPFLAGS
      if test -n "${_incdir}"; then
        CPPFLAGS="-I${_incdir} ${CPPFLAGS}"
