From 1f1c482482875522b61d835c46fd7afb9057ae3d Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 17 Dec 2018 15:38:07 -0600
Subject: [PATCH] pango: quickfix dep on glib?

---
 pango/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pango/meson.build b/pango/meson.build
index 099aed89..bb34be44 100644
--- a/pango/meson.build
+++ b/pango/meson.build
@@ -161,6 +161,7 @@ pkgconfig.generate(libpango,
   version: meson.project_version(),
   filebase: 'pango',
   subdirs: pango_api_name,
+  requires: glib_dep,
   install_dir: join_paths(pango_libdir, 'pkgconfig'),
 )
 
-- 
2.20.1

