From 91fdd674c316af0873d495f85f9c220b6e9e0132 Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@igalia.com>
Date: Wed, 3 Apr 2019 17:28:48 -0500
Subject: [PATCH 1/2] Remove workaround for old GnuTLS rehandshaking bug

---
 tls/tests/connection.c | 23 -----------------------
 1 file changed, 23 deletions(-)

diff --git a/tls/tests/connection.c b/tls/tests/connection.c
index 69fb286..5f25ea4 100644
--- a/tls/tests/connection.c
+++ b/tls/tests/connection.c
@@ -1685,17 +1685,6 @@ test_simultaneous_async (TestConnection *test,
   g_assert_cmpstr (test->buf, ==, TEST_DATA);
 }
 
-#ifdef BACKEND_IS_GNUTLS
-static gboolean
-check_gnutls_has_rehandshaking_bug (void)
-{
-  const char *version = gnutls_check_version (NULL);
-
-  return !strcmp (version, "3.6.1") ||
-         !strcmp (version, "3.6.2");
-}
-#endif
-
 static void
 test_simultaneous_async_rehandshake (TestConnection *test,
                                      gconstpointer   data)
@@ -1703,12 +1692,6 @@ test_simultaneous_async_rehandshake (TestConnection *test,
 #ifdef BACKEND_IS_OPENSSL
   g_test_skip ("this needs more research on openssl");
   return;
-#elif defined(BACKEND_IS_GNUTLS)
-  if (check_gnutls_has_rehandshaking_bug ())
-    {
-      g_test_skip ("test would fail due to https://gitlab.com/gnutls/gnutls/issues/426");
-      return;
-    }
 #endif
 
   test->rehandshake = TRUE;
@@ -1808,12 +1791,6 @@ test_simultaneous_sync_rehandshake (TestConnection *test,
 #ifdef BACKEND_IS_OPENSSL
   g_test_skip ("this needs more research on openssl");
   return;
-#elif defined(BACKEND_IS_GNUTLS)
-  if (check_gnutls_has_rehandshaking_bug ())
-    {
-      g_test_skip ("test would fail due to https://gitlab.com/gnutls/gnutls/issues/426");
-      return;
-    }
 #endif
 
   test->rehandshake = TRUE;
-- 
2.21.GIT

