From 5c165453ab1d0a91a0b7a5c8493bafddad4eaab2 Mon Sep 17 00:00:00 2001
From: Eric Freese <ericdfreese@gmail.com>
Date: Tue, 9 Apr 2019 12:34:04 -0600
Subject: [PATCH 2/2] 44214: <(...) substitutions shouldn't grab the terminal

---
 Src/exec.c | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/Src/exec.c b/Src/exec.c
index 79ef83c1e..6ac852112 100644
--- a/Src/exec.c
+++ b/Src/exec.c
@@ -4984,7 +4984,7 @@ getpipe(char *cmd, int nullexec)
 	procsubstpid = pid;
 	return pipes[!out];
     }
-    entersubsh(ESUB_PGRP, NULL);
+    entersubsh(ESUB_ASYNC|ESUB_PGRP, NULL);
     redup(pipes[out], out);
     closem(FDT_UNUSED, 0);	/* this closes pipes[!out] as well */
     cmdpush(CS_CMDSUBST);
-- 
2.21.GIT

