From 8adf5a4fc065f3fb1e2ede96706449116277cc69 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 25 Mar 2019 15:56:31 -0500
Subject: [PATCH] compton.c: differentiate above missing entirely vs destroying

---
 src/compton.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/compton.c b/src/compton.c
index b861f03..357a603 100644
--- a/src/compton.c
+++ b/src/compton.c
@@ -674,8 +674,12 @@ static void restack_win(session_t *ps, win *w, xcb_window_t new_above) {
 
 		bool found = false;
 		for (prev = &ps->list; *prev; prev = &(*prev)->next) {
-			if ((*prev)->id == new_above && (*prev)->state != WSTATE_DESTROYING) {
+			if ((*prev)->id == new_above) {
 				found = true;
+				if ((*prev)->state == WSTATE_DESTROYING) {
+					log_error("(%#010x, %#010x): new above window is WSTATE_DESTROYING", w->id, new_above);
+					new_above = XCB_NONE;
+				}
 				break;
 			}
 		}
-- 
2.21.GIT

