From d167cf48c2c4779a77d48031477ad46fffb747d3 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Thu, 28 Mar 2019 20:00:38 -0500
Subject: [PATCH] more logging

---
 src/win.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/win.c b/src/win.c
index 14f55a8..53f007d 100644
--- a/src/win.c
+++ b/src/win.c
@@ -956,6 +956,7 @@ void add_win(session_t *ps, xcb_window_t id, xcb_window_t prev) {
 		// already reparented elsewhere.
 		// BTW, we don't care about Input Only windows, except for stacking
 		// proposes, so we need to keep track of them still.
+		log_debug("Window %#010x: unviewable or unable to get attributes, ignoring.", id);
 		free(a);
 		return;
 	}
-- 
2.21.GIT

