From cf596656af362a8a415788894b282c546073d1b9 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Thu, 28 Mar 2019 23:59:34 -0500
Subject: [PATCH] glx: appease -fsanitize=integer

---
 src/backend/gl/glx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/backend/gl/glx.c b/src/backend/gl/glx.c
index 782fb4b..45e5495 100644
--- a/src/backend/gl/glx.c
+++ b/src/backend/gl/glx.c
@@ -70,7 +70,7 @@ struct glx_fbconfig_info *glx_find_fbconfig(Display *dpy, int screen, struct xvi
 	                          GLX_DRAWABLE_TYPE, GLX_PIXMAP_BIT,
 	                          GLX_X_VISUAL_TYPE, GLX_TRUE_COLOR,
 	                          GLX_X_RENDERABLE, true,
-	                          GLX_FRAMEBUFFER_SRGB_CAPABLE_EXT, GLX_DONT_CARE,
+	                          GLX_FRAMEBUFFER_SRGB_CAPABLE_EXT, (int)GLX_DONT_CARE,
 	                          GLX_BUFFER_SIZE, m.red_size + m.green_size +
 	                                           m.blue_size + m.alpha_size,
 	                          GLX_RED_SIZE, m.red_size,
-- 
2.21.GIT

