From 39e1a8ed8c4328f79da69f96a4d057de9c4389af Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Sun, 3 Feb 2019 14:53:07 -0600
Subject: [PATCH] resize_region: avoid resizing out of bounds, warn when
 happens

---
 src/render.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/render.c b/src/render.c
index 9855b32..761cbc1 100644
--- a/src/render.c
+++ b/src/render.c
@@ -765,8 +765,10 @@ static inline void resize_region(region_t *region, short mod) {
 	pixman_box32_t *rects = pixman_region32_rectangles(region, &nrects);
 	auto newrects = ccalloc(nrects, pixman_box32_t);
 	for (int i = 0; i < nrects; i++) {
-		int x1 = rects[i].x1 - mod;
-		int y1 = rects[i].y1 - mod;
+		int x1 = max_i(rects[i].x1 - mod, 0);
+		int y1 = max_i(rects[i].y1 - mod, 0);
+		//if (rects[i].x1 - mod < 0 || rects[i].y1 - mod < 0)
+		//	log_warn("damage resize created negative coord, using 0 instead");
 		int x2 = rects[i].x2 + mod;
 		int y2 = rects[i].y2 + mod;
 		int wid = x2 - x1;
-- 
2.20.GIT

