From b92b7295ac2eb0dd2bd175d510a36840d056b8a5 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Thu, 14 Mar 2019 01:58:19 -0500
Subject: [PATCH] beep

---
 src/win.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/win.c b/src/win.c
index 4f5aa6b..0c4829f 100644
--- a/src/win.c
+++ b/src/win.c
@@ -1281,10 +1281,12 @@ gen_by_val(win_extents);
 			// Note we only do this when screen is redirected, because
 			// otherwise win_data is not valid
 			assert(w->state != WSTATE_UNMAPPING && w->state != WSTATE_DESTROYING);
+			assert(w->win_image);
 			ps->backend_data->ops->release_image(ps->backend_data, w->win_image);
 			if (w->shadow_image) {
 				ps->backend_data->ops->release_image(ps->backend_data,
 				                                     w->shadow_image);
+				w->shadow_image = NULL;
 			}
 			auto pixmap = xcb_generate_id(ps->c);
 			xcb_composite_name_window_pixmap(ps->c, w->id, pixmap);
-- 
2.21.GIT

