From 351f08420c38ab11964cfba34c52a3b701817215 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 6 May 2019 01:16:53 -0500
Subject: [PATCH 1/2] escape "patterns", fix hyphen behavior but lose lua-regex

---
 z.lua | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/z.lua b/z.lua
index 5e4ef44..f201c0e 100755
--- a/z.lua
+++ b/z.lua
@@ -1253,6 +1253,9 @@ function path_match(pathname, patterns, matchlast)
 	return true
 end
 
+function escape_pattern(text)
+  return text:gsub("([^%w])", "%%%1")
+end
 
 -----------------------------------------------------------------------
 -- select matched pathnames
@@ -1263,7 +1266,8 @@ function data_select(M, patterns, matchlast)
 	local pats = {}
 	for i = 1, #patterns do
 		local p = patterns[i]
-		table.insert(pats, case_insensitive_pattern(p))
+		-- table.insert(pats, case_insensitive_pattern(p))
+		table.insert(pats, case_insensitive_pattern(escape_pattern(p)))
 	end
 	for i = 1, #M do
 		local item = M[i]
-- 
2.21.0

