From b93e1bceb93080f4380828e19b70df08d2d450e5 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 6 May 2019 01:26:58 -0500
Subject: [PATCH 2/2] escape just hyphens, as quickfix compromise

---
 z.lua | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/z.lua b/z.lua
index f201c0e..0dfe270 100755
--- a/z.lua
+++ b/z.lua
@@ -1257,6 +1257,10 @@ function escape_pattern(text)
   return text:gsub("([^%w])", "%%%1")
 end
 
+function escape_hyphen(text)
+  return text:gsub("%-", "%%-")
+end
+
 -----------------------------------------------------------------------
 -- select matched pathnames
 -----------------------------------------------------------------------
@@ -1267,7 +1271,7 @@ function data_select(M, patterns, matchlast)
 	for i = 1, #patterns do
 		local p = patterns[i]
 		-- table.insert(pats, case_insensitive_pattern(p))
-		table.insert(pats, case_insensitive_pattern(escape_pattern(p)))
+		table.insert(pats, case_insensitive_pattern(escape_hyphen(p)))
 	end
 	for i = 1, #M do
 		local item = M[i]
-- 
2.21.0

