Fix a construct rejected by gcc7.
diff --git a/aws-cpp-sdk-core/source/http/curl/CurlHttpClient.cpp b/aws-cpp-sdk-core/source/http/curl/CurlHttpClient.cpp
index c9e45b3..e6d4bab 100644
--- a/aws-cpp-sdk-core/source/http/curl/CurlHttpClient.cpp
+++ b/aws-cpp-sdk-core/source/http/curl/CurlHttpClient.cpp
@@ -344,7 +344,7 @@ size_t CurlHttpClient::ReadBody(char* ptr, size_t size, size_t nmemb, void* user
     HttpRequest* request = context->m_request;
     std::shared_ptr<Aws::IOStream> ioStream = request->GetContentBody();
 
-    if (ioStream != nullptr && size * nmemb)
+    if (ioStream != nullptr && size * nmemb != 0)
     {
         auto currentPos = ioStream->tellg();
         ioStream->seekg(0, ioStream->end);
