From 0c633e5a0a1fb2231e8eb2fc1e1c5d47d0d783c7 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Wed, 13 Feb 2019 07:10:37 -0600
Subject: [PATCH] bump BINPRM_BUF_SIZE from 128 to 1024

---
 binfmts.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/binfmts.h b/binfmts.h
index 4abad03..667179b 100644
--- a/binfmts.h
+++ b/binfmts.h
@@ -16,6 +16,7 @@ struct pt_regs;
 #define MAX_ARG_STRINGS 0x7FFFFFFF
 
 /* sizeof(linux_binprm->buf) */
-#define BINPRM_BUF_SIZE 128
+/* 1024 bytes ought to be enough for anyone */
+#define BINPRM_BUF_SIZE 1024
 
 #endif /* _UAPI_LINUX_BINFMTS_H */
-- 
2.21.0-rc0

