From f3cfe7794ac8bdce6d8596db02690c404c543131 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Fri, 1 Mar 2019 09:20:35 -0600
Subject: [PATCH] hack: replace tabs with spaces in Subject header

---
 src/callout.m      | 4 ++--
 src/message_file.m | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/callout.m b/src/callout.m
index 53481c7..5c49e43 100644
--- a/src/callout.m
+++ b/src/callout.m
@@ -252,7 +252,7 @@ parse_header(Key, unesc_string(Value), !Headers) :-
         !Headers ^ h_bcc := header_value(Value)
     ; Key = "Subject" ->
         % notmuch should provide the decoded value.
-        !Headers ^ h_subject := decoded_unstructured(Value)
+        !Headers ^ h_subject := decoded_unstructured(string.replace_all(Value, "\t", " "))
     ; Key = "Reply-To" ->
         !Headers ^ h_replyto := header_value(Value)
     ; Key = "References" ->
@@ -499,7 +499,7 @@ parse_thread(Json, Thread) :-
     ->
         TagSet = set.from_list(Tags),
         Thread = thread(thread_id(Id), timestamp(float(Timestamp)), Authors,
-            Subject, TagSet, Matched, Total)
+            string.replace_all(Subject, "\t", " "), TagSet, Matched, Total)
     ;
         notmuch_json_error
     ).
diff --git a/src/message_file.m b/src/message_file.m
index a3f2da0..405102b 100644
--- a/src/message_file.m
+++ b/src/message_file.m
@@ -67,7 +67,7 @@ read_headers(String, Pos0, Pos, !Headers) :-
         % Other headers should be decoded as well.
         ( strcase_equal(Name, "Subject") ->
             decode_unstructured(RawValue, Decoded),
-            Value = decoded_unstructured(Decoded)
+            Value = decoded_unstructured(string.replace_all(Decoded, "\t", " "))
         ;
             Value = header_value(RawValue)
         ),
-- 
2.21.0

