From 6af819cb6e907830ba89909b6d446440114c3ec9 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Thu, 18 Oct 2018 15:20:13 -0500
Subject: [PATCH 2/2] hide more tags that would be painful to mass-change

---
 src/tags.m | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/tags.m b/src/tags.m
index da8fb33..8000349 100644
--- a/src/tags.m
+++ b/src/tags.m
@@ -95,6 +95,15 @@ nondisplay_tag(tag("signed")).
 nondisplay_tag(tag("unread")).
 nondisplay_tag(tag("important")).
 nondisplay_tag(tag("inbox")).
+nondisplay_tag(tag("old/Research")).
+nondisplay_tag(tag("llvm dev lists/llvm-commits")).
+nondisplay_tag(tag("llvm dev lists/clang-commits")).
+nondisplay_tag(tag("llvm dev lists/clang-dev")).
+nondisplay_tag(tag("llvm dev lists/clang-users")).
+nondisplay_tag(tag("llvm dev lists/llvm-dev")).
+nondisplay_tag(tag("llvm dev lists/llvm-bugs")).
+nondisplay_tag(tag("_my projects/ALLVM")).
+nondisplay_tag(tag("_my projects/ALLVM/commits")).
 
 %-----------------------------------------------------------------------------%
 
-- 
2.19.1

