From b37b853ca8b8f5d133b5406f14b1ac0531be2055 Mon Sep 17 00:00:00 2001
From: Stian Soiland-Reyes <stain@apache.org>
Date: Mon, 13 Feb 2017 16:18:39 +0000
Subject: [PATCH 2/7] =?UTF-8?q?=F0=9F=94=92=20Good=20signature=20?=
 =?UTF-8?q?=F0=9F=94=93=20Bad=20signature?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 src/pager.m | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/pager.m b/src/pager.m
index 668b368..57e238f 100644
--- a/src/pager.m
+++ b/src/pager.m
@@ -1699,11 +1699,11 @@ draw_pager_line(Attrs, Panel, Line, IsCursor, !IO) :-
         BadAttr = GAttrs ^ bad_key,
         (
             Status = none,
-            draw(Panel, BadAttr, " ─• No signature ", !IO)
+            draw(Panel, BadAttr, " ─• 🔓No signature ", !IO)
         ;
             Status = good(MaybeFingerprint, _MaybeCreated, MaybeExpires,
                 MaybeUserId),
-            draw(Panel, GoodAttr, " ─• Good signature ", !IO),
+            draw(Panel, GoodAttr, " ─• 🔒 Good signature ", !IO),
             (
                 MaybeUserId = yes(UserId),
                 draw(Panel, BodyAttr, lstrip(UserId), !IO),
@@ -1737,13 +1737,13 @@ draw_pager_line(Attrs, Panel, Line, IsCursor, !IO) :-
             Status = not_good(BadStatus, MaybeKeyId),
             (
                 BadStatus = bad,
-                BadMessage = " ─• Bad signature "
+                BadMessage = " ─• 🔓 Bad signature "
             ;
                 BadStatus = error,
-                BadMessage = " ─• Error verifying signature "
+                BadMessage = " ─• 🔓 Error verifying signature "
             ;
                 BadStatus = unknown,
-                BadMessage = " ─• Problem verifying signature "
+                BadMessage = " ─• 🔓Problem verifying signature "
             ),
             draw(Panel, BadAttr, BadMessage, !IO),
             (
-- 
2.20.1

