From f20c3acec96cd57788c1c311f81045b682d28c8d Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Fri, 1 Mar 2019 11:03:28 -0600
Subject: [PATCH] use diff color, optionally, for 'total' part of count

---
 src/color.m      | 9 ++++++---
 src/index_view.m | 7 +++++--
 2 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/src/color.m b/src/color.m
index ba75084..2d49614 100644
--- a/src/color.m
+++ b/src/color.m
@@ -54,7 +54,8 @@
 :- type index_attrs
     --->    index_attrs(
                 i_generic       :: generic_attrs,
-                i_count         :: attr
+                i_count         :: attr,
+                i_counttotal    :: attr
             ).
 
 :- type thread_attrs
@@ -185,7 +186,8 @@ make_index_attrs(Config, GenericAttrs0, Attrs) :-
     Def = default_index_attrs(GenericAttrs),
     Sections = [Section, generic_section],
     cfg(Config, Def ^ i_count, Sections, "count", Count),
-    Attrs = index_attrs(GenericAttrs, Count).
+    cfg(Config, Def ^ i_counttotal, Sections, "counttotal", CountTotal),
+    Attrs = index_attrs(GenericAttrs, Count, CountTotal).
 
 :- pred make_thread_attrs(config::in, generic_attrs::in, status_attrs::in,
     thread_attrs::out) is det.
@@ -369,7 +371,8 @@ default_pager_attrs(GenericAttrs) =
 default_index_attrs(GenericAttrs0) =
     index_attrs(
         GenericAttrs0,
-        normal(green)
+        normal(green),
+        bold(black)
     ).
 
 :- func default_thread_attrs(generic_attrs, status_attrs) = thread_attrs.
diff --git a/src/index_view.m b/src/index_view.m
index 1407cbe..17acef9 100644
--- a/src/index_view.m
+++ b/src/index_view.m
@@ -1822,11 +1822,14 @@ draw_index_line(IAttrs, Panel, Line, _LineNr, IsCursor, !IO) :-
         25, Authors, ' ', !IO),
 
     ( Matched = Total ->
-        CountStr = format(" %3d     ", [i(Total)])
+        CountStr = format(" %3d", [i(Total)]),
+        TotalStr = "     "
     ;
-        CountStr = format(" %3d/%-3d ", [i(Matched), i(Total)])
+        CountStr = format(" %3d", [i(Matched)]),
+        TotalStr = format("/%-3d ", [i(Total)])
     ),
     mattr_draw(Panel, unless(IsCursor, IAttrs ^ i_count), CountStr, !IO),
+    mattr_draw(Panel, unless(IsCursor, IAttrs ^ i_counttotal), TotalStr, !IO),
 
     panel.getyx(Panel, Row, SubjectX0, !IO),
     mattr_draw(Panel, unless(IsCursor, Attrs ^ subject), Subject, !IO),
-- 
2.21.0

