diff --git a/fonts.conf b/fonts.conf
index 5e9daac..94a91ef 100644
--- a/fonts.conf
+++ b/fonts.conf
@@ -65,10 +65,11 @@
 		</edit>
 	</match>
 
-<!--
-  Load local system customization file
--->
-	<include ignore_missing="yes">conf.d</include>
+<!-- Other config files -->
+	<!-- upstream, including ~/.config/fontconfig/{fonts.conf,conf.d/} -->
+	<include>conf.d</include>
+	<!-- nix-specific and machine-specific -->
+	<include ignore_missing="yes">/etc/fonts-nix/fonts.conf</include>
 
 <!-- Font cache directory list -->
 
